helper/src/lib.rs

```diff
@@ -109,37 +91,26 @@ pub fn build_program_with_args(path: &str, args: BuildArgs) {
 /// Internal helper function to build the program with or without arguments.
 fn build_program_internal(path: &str, args: Option<BuildArgs>) {
   // Get the root package name and metadata.
-  let program_dir = std::path::Path::new(path);
+  let program_dir = Path::new(path);
   let metadata_file = program_dir.join("Cargo.toml");
   let mut metadata_cmd = cargo_metadata::MetadataCommand::new();
   let metadata = metadata_cmd.manifest_path(metadata_file).exec().unwrap();
   let root_package = metadata.root_package();
-  let root_package_name = root_package
-    .as_ref()
-    .map(|p| p.name.as_str())
-    .unwrap_or("Program");
+  let root_package_name = root_package.map(|p| p.name.as_str()).unwrap_or("Program");
 
   // Skip the program build if the ATHENA_SKIP_PROGRAM_BUILD environment variable is set to true.
   let skip_program_build = std::env::var("ATHENA_SKIP_PROGRAM_BUILD")
     .map(|v| v.eq_ignore_ascii_case("true"))
     .unwrap_or(false);
   if skip_program_build {
     println!(
-      "cargo:warning=Build skipped for {} at {} due to ATHENA_SKIP_PROGRAM_BUILD flag",
-      root_package_name,
-      current_datetime()
+      "cargo:warning=Skipping building {root_package_name} due to ATHENA_SKIP_PROGRAM_BUILD flag",
     );
     return;
   }
 
   // Activate the build command if the dependencies change.
   cargo_rerun_if_changed(&metadata, program_dir);
 
-  let _ = execute_build_cmd(&program_dir, args);
-
-  println!(
```

Comment:
Why is it useful? If it didn't fail, then it built the program. The build scripts should not emit anything on success to not pollute the terminal.

Reply:
> Why is it useful? If it didn't fail, then it built the program

It's how the user knows if the guest program was rebuilt or not. We inherited this from SP1 (see https://docs.succinct.xyz/writing-programs/compiling.html#build-script) and it feels useful to know this, without having to, e.g., check the timestamps of the ELF output files.

Reply:
It's an anti-pattern to print anything from build scripts. It should be assumed that if they ran successfully, they worked as intended. It's because build scripts are executed for **all** dependencies in the tree of dependencies and we don't want to see the output of a build script of every (implicit) dependency deep down the stack.

But if this is useful I will bring it back.

Reply:
Not disagreeing in general, but I do find it useful and reassuring to know that the guest build is working correctly. If you have a suggestion for a better way to do this, I'm all ears. I'm not going to block merging on this, but if we merge without this and I find that it's an issue later I'll PR a reversion of this change :P